app/views/play/level/PlayGameDevLevelView.coffee

```diff
@@ -186,12 +186,20 @@ module.exports = class PlayGameDevLevelView extends RootView
 
   onSurfaceTicked: (e) ->
     return if @studentGoals
-    goals = @surface.world?.thangMap?['Hero Placeholder']?.stringGoals
-    return unless _.size(goals)
-    @updateRealTimeGoals(goals)
+    # Set by users. Defined in `game.GameUI` component in the level editor.
+    if @world.uiText?.directions? and @world.uiText.directions.length
+      userDefinedGoals = @world.uiText.directions.map((directions) -> {type: "user_defined", directions})
+      @studentGoals = userDefinedGoals
+    else
+      @studentGoals = @world.thangMap['Hero Placeholder'].stringGoals
+      @studentGoals = @studentGoals?.map((g) -> JSON.parse(g))
+    # goals = @surface.world?.thangMap?['Hero Placeholder']?.stringGoals
+    return unless _.size(@studentGoals)
+    @updateRealTimeGoals()
 
   updateRealTimeGoals: (goals) ->
-    @studentGoals = goals?.map((g) -> JSON.parse(g))
+    if goals?
```

Comment:
Why are we adding this check?
Don't want to overwrite `@studentGoals`?

Reply:
Yeah. Only overwrite if goals are explicitly passed in.